src/pages/AddItemForm.js

```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+import { db } from '../lib/firebase';
+import { addDoc, collection } from 'firebase/firestore';
+// const formStyle = {
+//     display: 'block',
+//   };
+async function addToDb(item_name, purchase_interval, user_token) {
+  try {
+    const docRef = await addDoc(collection(db, 'groceries'), {
+      item_name: item_name,
+      purchase_interval: purchase_interval,
+      user_token: user_token,
+      last_purchased_date: null,
+    });
+
+    console.log('Document written with ID: ', docRef.id);
+  } catch (e) {
+    console.error('Error adding document: ', e);
+  }
+}
+
+class AddItemForm extends React.Component {
+  constructor(props) {
+    //super is a method from React (create a class from a parent class)
+    super();
+    this.state = {
+      item_name: '',
+      purchase_interval: 7,
+      user_token: props.token.token,
+    };
+    //.bind(this) -The bind() is an inbuilt method in React that is used to pass the data as an argument to the function of a class based component
+    this.handleNameChange = this.handleNameChange.bind(this);
+    this.handleRadioChange = this.handleRadioChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+  //input
+  handleNameChange = (e) => {
+    this.setState({ item_name: e.target.value });
+  };
+  handleRadioChange = (e) => {
+    this.setState({
+      purchase_interval: e.target.value,
+    });
+  };
+  handleSubmit = async (e) => {
+    e.preventDefault();
+    console.log(this.state);
+    addToDb(
+      this.state.item_name,
+      this.state.purchase_interval,
+      this.state.user_token,
+    );
+    alert('Item was submitted: ' + this.state.item_name);
+  };
+  //TODO: check wireframe/accessibility requirements
+  render() {
+    return (
+      <form onSubmit={this.handleSubmit}>
+        <label htmlFor="item_name">
+          Item Name:
+          <input
+            type="text"
+            value={this.state.item_name}
+            onChange={this.handleNameChange}
+          />
+        </label>
+        <p>How soon will you buy this again?</p>
+        <fieldset>
+          <label htmlFor="soon">
```

Comment:
I noticed that nothing is selected when we first navigate to the page. I think that this is because the default state is a number, whereas we're comparing to a string. So, I think that we want to make the default state into a string. (i.e. const [purchaseIntervale, setPurchaseInterval] = useState("7");

Reply:
I was aiming to send an integer to the DB because of the AC, but I changed the value in the wrong place. Thanks for catching that! I just changed it to an integer in the DB call instead.

Reply:
Nice! Figuring out and deciding types for numbers like this is super duper tough and there are usually lots of right ways to do it.